python/rmm/_lib/memory_resource_util.hpp

```diff
@@ -0,0 +1,191 @@
+#include <memory>
+#include <rmm/mr/device/cnmem_managed_memory_resource.hpp>
+#include <rmm/mr/device/cnmem_memory_resource.hpp>
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+#include <rmm/mr/device/fixed_multisize_memory_resource.hpp>
+#include <rmm/mr/device/fixed_size_memory_resource.hpp>
+#include <rmm/mr/device/hybrid_memory_resource.hpp>
+#include <rmm/mr/device/logging_resource_adaptor.hpp>
+#include <rmm/mr/device/managed_memory_resource.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+#include <string>
+#include <vector>
+
+class device_memory_resource {
+ public:
+  virtual std::shared_ptr<rmm::mr::device_memory_resource> get_mr() = 0;
+};
+
+class cuda_memory_resource : public device_memory_resource {
+ public:
+  cuda_memory_resource() : mr(std::make_shared<rmm::mr::cuda_memory_resource>()) {}
+
+  std::shared_ptr<rmm::mr::device_memory_resource>
+  get_mr() {
+    return mr;
+  }
+
+ private:
+  std::shared_ptr<rmm::mr::cuda_memory_resource> mr;
```

Comment:
Need to add a note explaining why these are defined in each of the derived classes instead of in the base class.